quake.6
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH QUAKE 6 "2016-01-16"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
quake \- classic first person shooter
.SH SYNOPSIS
.B quake
.RI [ WRAPPER-OPTIONS ]
.RI [ ENGINE-OPTIONS ]
.RI [ ENGINE-COMMANDS ] ...
.br
.SH DESCRIPTION
This manual page documents briefly the
.B quake
command.
.PP
.\" TeX users may be more comfortable with the \fB<whatever>\fP and
.\" \fI<whatever>\fP escape sequences to invode bold face and italics,
.\" respectively.
\fBQuake\fP is a popular first-person shooter game that appeared in 1996,
released by iD Software.  This wrapper script will launch the Quake engine.  The
game data needs to be installed independently using the 'game-data-packager'
program, and by default it will be looked for at \fI/usr/share/games/quake\fR.
.PP
This wrapper script accepts the two common GNU-style options to get information
on the program.  Any further arguments are passed to the engine unmodified.
Consult the engine documentation for full details, but most engines support a
syntax whereby engine switches are prefixed with a minus sign, and you can
give console commands by prefixing with a plus sign.  See the examples section
for some common cases.
.PP
Additionally, if \fBquake\fR is run via a symbolic link whose name contains
\fBarmagon\fR, \fBhipnotic\fR, \fBdissolution\fR or \fBrogue\fR,
it will play one of the Quake mission packs, \fBScourge of Armagon\fR
(by Hipnotic Software) or \fBDissolution of Eternity\fR (by Rogue
Entertainment). This requires that additional data files are
installed.
.SH OPTIONS
These options are accepted for \fBWRAPPER-OPTIONS\fR in the
synopsis.
.TP
.B \-h, \-\-help
Show summary of options.
.TP
.B \-v, \-\-version
Show version of program.
.SH EXAMPLES
These are all engine dependent.
.TP
Play the Zerstoerer mod:
.B quake -game zer
.TP
Play a specific map using the Drake expansion:
.B quake -game drake +map sludge1
.TP
Connect to a server for multiplayer:
.B quake +connect dm.quakeone.net:27000
.SH SEE ALSO
.BR quake-server (6),
.BR quakespasm (6),
.BR game-data-packager (6).
.SH AUTHOR
This manual page, and the Quake wrapper script, were written by David Banks
<amoebae@gmail.com>, for the Debian project (and may be used by others).  The
package was based on the \fBquake3\fR package by Simon McVittie.


debian/changelog
game-data-packager (44) unstable; urgency=medium

  New game support:

  [ Alexandre Detiste ]
  * Enemy Territory: Quake Wars, using the proprietary x86 binaries
  * ScummVM games:
    - Amazon: Guardians of Eden
    - Beavis and Butthead in Virtual Stupidity
    - The Bizarre Adventures of Woodruff and the Schnibble
    - Conquests of Camelot: The Search for the Grail
    - Conquests of the Longbow: The Legend of Robin Hood
    - Freddy Pharkas: Frontier Pharmacist
    - Hoyle's Official Book of Games
    - Jones in the Fast Lane
    - Leather Goddesses of Phobos 2
    - Mixed-Up Mother Goose
    - Nippon Safes Inc.
    - Once Upon A Time: Little Red Riding Hood
    - Pajama Sam series
    - Rex Nebular and the Cosmic Gender Bender
    - Ringworld 1 & 2
    - Rodney's Funscreen
    - The Lost Files of Sherlock Holmes: The Case of the Rose Tattoo
    - Simon the Sorcerer's Puzzle Pack
    - Tony Tough and the Night of Roasted Moths
    - Touché: The Adventures of the Fifth Musketeer
    - Troll's Tale
    - Urban Runner
    - Ween: The Prophecy
    - Winnie the Pooh in the Hundred Acre Wood
  * Douglas Adams's "Bureaucracy" text game
  * The Elder Scrolls III: Morrowind (Closes: #776541)
  * Ultima VII (Closes: #784733)

  [ Stephen Kitt ]
  * Infocom games:
    - A Mind Forever Voyaging
    - Ballyhoo
    - Border Zone
    - Cutthroats
    - Deadline
    - Enchanter
    - Hollywood Hijinx
    - Infidel
    - Leather Goddesses of Phobos
    - The Lurking Horror
    - Moonmist
    - Nord and Bert Couldn't Make Head or Tail of It
    - Plundered Hearts
    - Seastalker
    - Sherlock: The Riddle of the Crown Jewels
    - Sorcerer
    - Spellbreaker
    - Starcross
    - Stationfall
    - Suspect
    - Suspended
    - Trinity
    - Wishbringer
    - Witness

  [ Simon McVittie ]
  * Unreal (classic or Gold), using the proprietary x86 binaries

  Support for new languages and versions:

  [ Alexandre Detiste ]
  * Discworld 1: add de/fr/es/it floppy version
  * RTCW: add Italian version, Thanks to iortcw upstream

  [ Stephen Kitt ]
  * Cruise for a Corpse: add French CD version
  * The Dig: add English CD version

  Enhancements and bug fixes:

  [ Alexandre Detiste ]
  * fix en_GB auto-detection
  * fix autodownloading of GrimFandango Italian demo
  * fix 'gog', 'atlantis' & 'baldurs-gate-2' targets (Closes: #804804)
  * use steamcmd to download Linux & Windows game assets (Closes: #775365)
  * re-add Doom 2 - MasterLevels launcher icon that got lost in v42 or v43

  [ Simon McVittie ]
  * runtime: add a generic Gtk launcher, initially for Unreal
  * Add an experimental AppArmor profile for the non-free Unreal binaries,
    to protect the rest of the system from any exploits

  [ Stephen Kitt ]
  * Update the GOG installer for The Dig (English).
  * Add Sam & Max CD manual.

  Internal changes:

  [ Alexandre Detiste ]
  * fedora: prepare initial release
  * make install: fix inclusion of game_data_packager/unpack in .deb
  * man pages: install man pages in Makefile
  * refactor packaging for various games
  * avoid computing known md5 again in our dh_md5sums implementation
  * steam: semi-automatically tag remaining native games

  [ Simon McVittie ]
  * quake, quake2: make aliases consistently strings
  * Always build vfs.zip, and optionally use it even when uninstalled
  * to_yaml: turn sets into sorted lists, and output more fields
  * Add a test that various forms of the game data are equivalent
  * GameData: do not crash if _iter_expand_groups changes self.files
  * build: allow tar.* as an unpacker format
  * build: add support for xdelta version 1 patches
  * Add an abstraction and limited auto-detection for archive unpacking
  * Add library code to extract Unreal mods and installers
  * make-template: always capture SHA1 and SHA256: we can edit them out
    later if desired, but if the g-d-p maintainers don't own a particular
    game, more information is better than less
  * make-template: use groups
  * make-template: refactor to support auto-detection and recursion into
    (some) archives
  * make-template: preserve case for files from InnoExtract
  * make-template: preserve full name of license files
  * Add support for forcing a package's short and long description
  * Allow packages to be marked as deliberately empty
  * Generalize support for $install_to in symlinks to cover $assets, etc.
  * Allow any file to be made executable by the installation process
  * Do not try to obtain non-preferred alternatives for files
  * Accept $GDP_DEBUG as a synonym for $DEBUG
  * Switch Vcs-Git to https (see #810378)

 -- Simon McVittie <smcv@debian.org>  Fri, 22 Jan 2016 23:01:25 +0000

game-data-packager (43) unstable; urgency=medium

  New game support:

  [ Alexandre Detiste ]
  * X-COM: UFO Defense (Closes: #793017)
  * ScummVM games:
    - 3 Skulls of the Toltecs
    - Bargon Attack
    - The Black Cauldron
    - Blue Force
    - Castle of Dr. Brain & The Island of Dr. Brain
    - Chivalry is Not Dead
    - Codename: ICEMAN
    - Cruise for a Corpse
    - EcoQuest 1 & 2
    - Enclosure (fan-made game)
    - Eye of the Beholder 1 & 2
    - The Feeble Files
    - Future Wars (English version)
    - Gold Rush!
    - Hank's Quest (fan-made game)
    - Hopkins FBI
    - Hugo's House of Horrors and its two sequels
    - Lands of Lore: The Throne of Chaos
    - Laura Bow 1 & 2
    - The Lost Files of Sherlock Holmes: The Case of the Serrated Scalpel
       (support for this game is currently being added to scummvm)
    - Lost in Time
    - Manhunter 1 & 2
    - The Manhole
    - The Neverhood
    - Personal Nightmare
    - Slater & Charlie Go Camping
    - Toonstruck

  [ Markus Koschany ]
  * Wolfenstein: Enemy Territory, for the etlegacy engine (Closes: #780446)
  * Baldur's Gate - The Original Saga (GOG.com English Linux installer)
    for the gemrb engine (Closes: #776540)
  * Baldur's Gate 2 (GOG.com English Linux installer) for the gemrb engine
    (Closes: #800712)

game_data_packager/games/quake.py
#!/usr/bin/python3
# encoding=utf-8
#
# Copyright © 2015 Simon McVittie <smcv@debian.org>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
#
# You can find the GPL license text on a Debian system under
# /usr/share/common-licenses/GPL-2.

import logging
import os

from .. import GameData
from ..build import (PackagingTask)
from ..paths import DATADIR
from ..util import TemporaryUmask, mkdir_p

logger = logging.getLogger('game-data-packager.games.quake')

class QuakeTask(PackagingTask):
    """With hindsight, it would have been better to make the TryExec
    in quake point to a symlink to the quake executable, or something;
    but with a name like hipnotic-tryexec.sh it would seem silly for it
    not to be a shell script.
    """

    def get_control_template(self, package):
        for name in (package.name, 'quake-common'):
            path = os.path.join(DATADIR, '%s.control.in' % name)
            if os.path.exists(path):
                return path
        else:
            raise AssertionError('quake-common.control.in should exist')

    def modify_control_template(self, control, package, destdir):
        super(QuakeTask, self).modify_control_template(control, package,
                destdir)

        desc = control['Description']
        desc = desc.replace('LONG', (package.longname or self.game.longname))
        control['Description'] = desc

    def fill_extra_files(self, package, destdir):
        super(QuakeTask, self).fill_extra_files(package, destdir)

        for path in package.install:
            if path.startswith('hipnotic'):
                detector = 'hipnotic-tryexec.sh'
                break
            elif path.startswith('rogue'):
                detector = 'rogue-tryexec.sh'
                break
        else:
            return

        with TemporaryUmask(0o022):
            quakedir = os.path.join(destdir, 'usr/share/games/quake')
            mkdir_p(quakedir)
            path = os.path.join(quakedir, detector)
            with open(path, 'w') as f:
                f.write('#!/bin/sh\nexit 0\n')
            os.chmod(path, 0o755)

class QuakeGameData(GameData):
    def construct_task(self, **kwargs):
        return QuakeTask(self, **kwargs)

    def add_parser(self, parsers, base_parser):
        parser = super(QuakeGameData, self).add_parser(parsers, base_parser,
                conflict_handler='resolve')
        parser.add_argument('-m', '-d', dest='packages', action='append_const',
                const='quake-registered',
                help='Equivalent to --package=quake-registered')
        parser.add_argument('-s', dest='packages', action='append_const',
                const='quake-shareware',
                help='Equivalent to --package=quake-shareware')
        parser.add_argument('--mp1', '-mp1', dest='packages',
                action='append_const', const='quake-armagon',
                help='Equivalent to --package=quake-armagon')
        parser.add_argument('--mp2', '-mp2', dest='packages',
                action='append_const', const='quake-dissolution',
                help='Equivalent to --package=quake-dissolution')
        parser.add_argument('--music', dest='packages',
                action='append_const', const='quake-music',
                help='Equivalent to --package=quake-music')
        parser.add_argument('--mp1-music', dest='packages',
                action='append_const', const='quake-armagon-music',
                help='Equivalent to --package=quake-armagon-music')
        parser.add_argument('--mp2-music', dest='packages',
                action='append_const', const='quake-dissolution-music',
                help='Equivalent to --package=quake-dissolution-music')
        return parser

GAME_DATA_SUBCLASS = QuakeGameData
